api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -288,6 +288,62 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() {
 		assertTrue(concept.getSet());
 	}
 	
+	/**
+	 * @see ConceptServiceImpl#saveDrug(Drug) 
+	 */
+	@Test
+	public void saveDrug_shouldPutGeneratedIdOntoReturnedDrug() {
+		Drug drug = new Drug();
+		Concept concept = new Concept();
+		concept.addName(new ConceptName("Concept", new Locale("en", "US")));
+		concept.addDescription(new ConceptDescription("Description", new Locale("en", "US")));
+		concept.setConceptClass(new ConceptClass(1));
+		concept.setDatatype(new ConceptDatatype(1));
+		Concept savedConcept = conceptService.saveConcept(concept);
+		drug.setConcept(savedConcept);
+		assertNull(drug.getDrugId());
+		Drug savedDrug = conceptService.saveDrug(drug);
+		assertNotNull(savedDrug.getDrugId());
+	}
+
+	/**
+	 * @see ConceptServiceImpl#saveDrug(Drug)
+	 */
+	@Test
+	public void saveDrug_shouldCreateNewDrugInDatabase() {
+		Drug drug = new Drug();
+		Concept concept = new Concept();
+		concept.addName(new ConceptName("Concept", new Locale("en", "US")));
+		concept.addDescription(new ConceptDescription("Description", new Locale("en", "US")));
+		concept.setConceptClass(new ConceptClass(1));
+		concept.setDatatype(new ConceptDatatype(1));
+		Concept savedConcept = conceptService.saveConcept(concept);
+		drug.setConcept(savedConcept);
+		Drug savedDrug = conceptService.saveDrug(drug);
+		assertNotNull(conceptService.getDrug(savedDrug.getDrugId()));
```

Comment:
Shouldn't we check it for null before saving drugs? Then how can we ensure that it's newly created?